src/lib/utils.js

```diff
@@ -369,3 +369,90 @@ export function getSharedStyleFromURL(mapId) {
 
   return style ? +style : null;
 }
+
+export function getCategoryForLayers(layersToOpenCategory, categories) {
+  const  getMenuGroupMapLayers = (groupName, children) => {
+    const subGroups = children.filter(child => !child.id);
+  
+    if (subGroups.length) {
+      let visibleLayerIds = [];
+  
+      subGroups.forEach(sg => {
+        Object.keys(sg).forEach(key => {
+          const subGroupVisibleLayerIds = getMenuGroupMapLayers(groupName, sg[key].layers);
+          visibleLayerIds = [...visibleLayerIds, ...subGroupVisibleLayerIds];
+        });
+      });
+  
+      // Return all layer Ids of the nested subgroups
+      return visibleLayerIds;
+    } else {
+      // Return the Ids of layers for the group
+      return children.map(child => child.id);
+    }
+  }
+  const layerCategory = [];
+  if (layersToOpenCategory && categories) {
+    layersToOpenCategory
+      .filter(l => !l.isCatOpen)
```

Comment:
`layersToOpenCategory` should be an array of layer id strings not object. `.filter(l => !l.isCatOpen)` is not required here